agent/grpc-external/services/peerstream/stream_tracker.go

```diff
@@ -136,6 +137,39 @@ func (t *Tracker) DeleteStatus(id string) {
 	delete(t.streams, id)
 }
 
+// IsHealthy is a calculates the health of a peering status.
+// We define a peering as unhealthy if its status has been in the following
+// states for longer than the configured incomingHeartbeatTimeout.
+//   - If it is disconnected
+//   - If the last received Nack is newer than last received Ack
+//   - If the last received error is newer than last received success
+//
+// If none of these conditions apply, we call the peering healthy.
+func (t *Tracker) IsHealthy(s Status) bool {
```

Comment:
I moved `IsHealthy` to be a method under `Tracker` because some of the calculations require dependencies which didn't make sense to live in a data object. If we consider `Status` as a bundle of data points, it should be the `Tracker`'s responsibility to interpret it as healthy or not.